src/common/runtime/wpt.ts

```diff
@@ -17,19 +22,60 @@ declare function setup(properties: { explicit_done?: boolean }): void;
 declare function promise_test(f: (t: WptTestObject) => Promise<void>, name: string): void;
 declare function done(): void;
 
+declare let loadWebGPUExpectations: Promise<RawTestQueryStringWithExpectation[]> | undefined;
+
 setup({
   // It's convenient for us to asynchronously add tests to the page. Prevent done() from being
   // called implicitly when the page is finished loading.
   explicit_done: true,
 });
 
 (async () => {
+  const workerEnabled = optionEnabled('worker');
+  const worker = workerEnabled ? new TestWorker(false) : undefined;
+
   const loader = new DefaultTestFileLoader();
   const qs = new URLSearchParams(window.location.search).getAll('q');
   assert(qs.length === 1, 'currently, there must be exactly one ?q=');
-  const testcases = await loader.loadCases(parseQuery(qs[0]));
+  const testcaseQuery = parseQuery(qs[0]);
+  const testcases = await loader.loadCases(testcaseQuery);
+
+  const expectations: TestQueryWithExpectation[] = [];
+  if (typeof loadWebGPUExpectations !== 'undefined') {
+    (await loadWebGPUExpectations).forEach(entry => {
+      const url = new URL(`${window.location.origin}/${entry.query}`);
```

Comment:
Can you add a comment with an example of what this URL will look like? at first glance it looks like it'll be `http://localhost:8080/webgpu:examples:*` but that appears not to be the case.

Reply:
Added a jsdoc comment. It's not super great..? But the format is slightly different in/out of WPT.